api/envoy/config/route/v3/route_components.proto

```diff
@@ -1087,6 +1087,27 @@ message RetryPolicy {
     google.protobuf.Duration max_interval = 2 [(validate.rules).duration = {gt {}}];
   }
 
+  message RateLimitedRetryBackOff {
+    // Specifies the rate limited retry reset headers (like ``Retry-After`` or
+    // ``X-RateLimit-Reset``) to match against the response. If a request is
+    // being retried and the response contains one of these headers the value
+    // of the header will be used as the retry back off interval, with jitter
+    // added. If multiple headers are specified they are tried in the order
+    // given and the first matching header in the response is used.
+    //
+    // The value of the header must be either:
+    //  1. An interval in seconds as an integer, for example: ``120``.
+    //  2. A unix timestamp as an integer, for example: ``1595320702``. The timestamp
+    //     must be in the future relative to the current system time.
```

Comment:
How can you differentiate between an interval and a unix time stamp? Doesn't there need to be some other format included or some configuration to tell what type of format the header is? It seems like maybe it should be a repeated list of messages where each message includes a HeaderMatcher and then there is a format enum or something?

Also, can you clarify in the docs what happens if the format is invalid?

Reply:
The `Retry-After` [spec](https://tools.ietf.org/html/rfc7231#section-7.1.3) allows `http-date` as a format but not a unix timestamp.

The `X-Ratelimit-Reset` [draft spec ](https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html#rfc.section.3.3) allows IMF-fixdate, but it is not recommended. Also doesn't specify unix timestamp as a possibility.

Reply:
Hi @Pchelolo ! 

Yes, that's correct. I'm aware of the standards, however in the wild they are not followed as written. I have surveyed a number of public APIs which you can find below and it's based on this that I've proposed to support the above two formats:
https://docs.google.com/document/d/1NSzrx3-KsAFs0ObaLQZUVIt9O4PZv3mXi3Lm7LNzmZE/edit#heading=h.nd3aike2c678

Reply:
My initial idea was that we could tell whether the value in the header represents a number of seconds or a Unix timestamp. For it to be a valid Unix timestamp in a retry context it needs to be greater than the current system time. @zuercher convinced me that we should instead have a dedicated protobuf message that captures both pieces of information: a) the header name and b) the format